src/transformation/visitors/switch.ts

```diff
@@ -18,10 +18,13 @@ export const transformSwitchStatement: FunctionVisitor<ts.SwitchStatement> = (st
 
     let statements: lua.Statement[] = [];
 
-    const caseClauses = statement.caseBlock.clauses.filter(ts.isCaseClause);
-
     // Starting from the back, concatenating ifs into one big if/elseif statement
-    const concatenatedIf = caseClauses.reduceRight((previousCondition, clause, index) => {
+    const concatenatedIf = statement.caseBlock.clauses.reduceRight((previousCondition, clause, index) => {
```

Comment:
Not, related to this PR. But I find this reduceRight call extremely hard to read. If you want to you can refactor it. maybe by splitting the generation of the statements and the concatenation of the statements?
```ts
const ifStatementsForCases = statement.caseBlock.clauses.filter(!ts.isDefaultClause(clause)).map(
    ...
    return lua.createIfStatement(condition, lua.createBlock([goto])));
);
const concatenatedIf = ifStatementsForCases.reduceRight((prev, ifStatement) => {
    prev.elseBlock = ifStatement;
    return ifStatement;
});
```

If not, I am fine with merging the PR as it is.